sunpy/net/helioviewer.py

```diff
@@ -450,3 +433,18 @@ def _request(self, params):
     def _format_date(self, date):
         """Formats a date for Helioviewer API requests"""
         return parse_time(date).isot + "Z"
+
+    def _get_source_id(self, source_id, key):
```

Comment:
Why is source_id an input?

Reply:
Because of the same reason that if a user enters the source_id which is allowed as of now then we don't want to calculate the source_id using _get_source_id function and just return the source_id entered by the user

Reply:
But this is meant to get the source_id, I don't see why we should use this to check it.

Reply:
But we do need to check it somewhere. We can check it in the calling function